.eslintrc

```diff
@@ -1,19 +1,37 @@
 {
+  "env": {
+    "browser": true,
+    "es6": true,
+    "jest": true
+  },
+  "globals": {
+    "Atomics": "readonly",
+    "SharedArrayBuffer": "readonly"
+  },
   "root": true,
   "parser": "@typescript-eslint/parser",
   "parserOptions": {
     "ecmaVersion": 6,
     "sourceType": "module",
     "ecmaFeatures": {
       "jsx": true
-    }
+    },
+    "project": "./tsconfig.json"
   },
-  "plugins": ["@typescript-eslint"],
+  "plugins": ["react", "@typescript-eslint", "jest"],
   "extends": [
-    "react-app",
+    "airbnb-typescript",
+    "airbnb/hooks",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:jest/recommended",
+    "plugin:prettier/recommended",
     "plugin:react/recommended",
     "plugin:@typescript-eslint/eslint-recommended",
-    "plugin:react-hooks/recommended"
+    "plugin:react-hooks/recommended",
+    "prettier",
```

Comment:
My ideal tweaks to our current `prettier` config:

- https://prettier.io/docs/en/options.html
- https://prettier.io/docs/en/options.html#semicolons
  - turn this off
- https://prettier.io/docs/en/options.html#quotes
  - turn this on
- https://prettier.io/docs/en/options.html#jsx-quotes
  - turn this on
- https://prettier.io/docs/en/options.html#trailing-commas
  - es5
- https://prettier.io/docs/en/options.html#arrow-function-parentheses
  - i prefer 'omit when possible' for this.. though that was in plain JS. If we need to manually add type annotations then this would be ok, but since a lot of the time we can allow it to infer, I still think I prefer avoiding them when possible

Reply:
Sounds good! Would you be OK to add those after we land this PR? Also, how can we make sure prettier's autoformat-on-save for vscode will honour these tweaks?